### PR TITLE
Preparations before our own name minifier.

### PR DESCRIPTION
**Pull request by sjrd**
First few commits extracted from #4930, as I believe they're good progress on their own, and they should be mergeable at this point.

### CODE REVIEW

File: linker/shared/src/main/scala/org/scalajs/linker/backend/emitter/SJSGen.scala [hunk 1]
```diff
@@ -113,10 +113,10 @@ private[emitter] final class SJSGen(
   }
 
   def usesUnderlyingTypedArray(elemTypeRef: NonArrayTypeRef): Boolean =
-    getArrayUnderlyingTypedArrayClassRef(elemTypeRef)(Position.NoPosition).nonEmpty
+    getArrayUnderlyingTypedArrayClassRef(elemTypeRef)(GlobalRefTracking.Dangerous, Position.NoPosition).nonEmpty
```
Comment (gzm0) [inline linker/shared/src/main/scala/org/scalajs/linker/backend/emitter/SJSGen.scala:116]: Why is the tracking not implicitly provided here? Probably worth a comment?

File: linker/shared/src/main/scala/org/scalajs/linker/backend/emitter/VarGen.scala [hunk 2]
```diff
@@ -39,6 +39,8 @@ private[emitter] final class VarGen(jsGen: JSGen, nameGen: NameGen,
   import jsGen._
   import nameGen._
 
+  private implicit val globalRefTracking = config.topLevelGlobalRefTracking
```
Comment (gzm0) [inline linker/shared/src/main/scala/org/scalajs/linker/backend/emitter/VarGen.scala:42]: Add a comment why this is OK? (IIUC we only ever generate dangerous globalRefs in here).

Is there a fundamental reason not to wire this through with implicits? I realize it is more boilerplate-y in this case, but would be IMO less surprising?

**Review by gzm0 — CHANGES_REQUESTED** [review]
Nice! Just two questions RE tracking specifics.

File: linker/shared/src/main/scala/org/scalajs/linker/backend/emitter/SJSGen.scala (hunk 1 above)
Comment (sjrd) [reply linker/shared/src/main/scala/org/scalajs/linker/backend/emitter/SJSGen.scala]: Comment added.

File: linker/shared/src/main/scala/org/scalajs/linker/backend/emitter/VarGen.scala (hunk 2 above)
Comment (sjrd) [reply linker/shared/src/main/scala/org/scalajs/linker/backend/emitter/VarGen.scala]: Huh good point. Probably I did that as a temporary measure while refactoring. I switched to implicits in individual methods.

File: linker/shared/src/main/scala/org/scalajs/linker/backend/emitter/SJSGen.scala [hunk 3]
```diff
@@ -112,11 +112,18 @@ private[emitter] final class SJSGen(
         args.toList)
   }
 
-  def usesUnderlyingTypedArray(elemTypeRef: NonArrayTypeRef): Boolean =
-    getArrayUnderlyingTypedArrayClassRef(elemTypeRef)(Position.NoPosition).nonEmpty
+  def usesUnderlyingTypedArray(elemTypeRef: NonArrayTypeRef): Boolean = {
+    /* We are only interested in whether `getArrayUnderlyingTypedArrayClassRef`
+     * returns a `Some` or not. We do not keep the result, so the `Position`
+     * and the `GlobalRefTracking` are irrelevant.
+     */
```
Comment (gzm0) [inline linker/shared/src/main/scala/org/scalajs/linker/backend/emitter/SJSGen.scala:119]: :facepalm: I could have figured that out. But the comment for sure is still a good idea :P

**Review by gzm0 — APPROVED** [review]

**Pull request merged**